### PR TITLE
[Fix](Nereids) ntile function should check argument

### PR DESCRIPTION
**Pull request by LiBinfeng-01**
## Proposed changes

Problem:
when ntile using 0 as parameter, be would core because no checking of parameter
Solved:
check parameter in fe analyze

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by LiBinfeng-01** [comment]
run buildall

**Comment by LiBinfeng-01** [comment]
run buildall

**Comment by LiBinfeng-01** [comment]
run buildall

**Comment by LiBinfeng-01** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/window/Ntile.java [hunk 1]
```diff
@@ -64,6 +67,27 @@ public Ntile withChildren(List<Expression> children) {
         return new Ntile(children.get(0));
     }
 
+    @Override
+    public void checkLegalityBeforeTypeCoercion() {
+        DataType type = getBuckets().getDataType();
+        if (!type.isNumericType()) {
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/window/Ntile.java:73]: isIntegralType ?

**Comment by LiBinfeng-01** [comment]
run buildall

**Review by morrySnow — APPROVED** [review]

**Review by 924060929 — APPROVED** [review]

**Pull request merged**